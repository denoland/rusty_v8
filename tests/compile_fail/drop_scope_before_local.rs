// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.
use std::pin::pin;

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut scope_pinned = pin!(v8::HandleScope::new(&mut isolate));
  let mut scope = scope_pinned.as_mut().init();

  let local = v8::Integer::new(&mut scope, 123);
  drop(scope_pinned);

  local.is_int32();
}

fn mock<T>() -> T {
  unimplemented!()
}
