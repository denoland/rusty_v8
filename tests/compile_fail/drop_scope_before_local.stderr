error[E0505]: cannot move out of `scope_pinned` because it is borrowed
  --> tests/compile_fail/drop_scope_before_local.rs:10:8
   |
6  |   let mut scope_pinned = pin!(v8::HandleScope::new(&mut isolate));
   |       ---------------- binding `scope_pinned` declared here
7  |   let mut scope = scope_pinned.as_mut().init();
   |                   ------------ borrow of `scope_pinned` occurs here
...
10 |   drop(scope_pinned);
   |        ^^^^^^^^^^^^ move out of `scope_pinned` occurs here
11 |
12 |   local.is_int32();
   |   ----- borrow later used here
